frontend/test/pytest/test_adjoint.py

```diff
@@ -17,566 +17,1652 @@
-        C_adjoint(qml.PauliY(val))
-        C_adjoint(qml.RZ(theta, wires=val))
-        return qml.state()
-
-    @qml.qnode(device)
-    def PL_workflow(theta, val):
-        PL_adjoint(qml.RX(jnp.pi, val))
-        PL_adjoint(qml.PauliY(val))
-        PL_adjoint(qml.RZ(theta, wires=val))
-        return qml.state()
-
-    actual = C_workflow(theta, val)
-    desired = PL_workflow(theta, val)
-    assert_allclose(actual, desired, atol=1e-6, rtol=1e-6)
-
-
-@pytest.mark.parametrize("w, p", [(0, 0.5), (0, -100.0), (1, 123.22)])
-def test_adjoint_param_fun(w, p, backend):
-    """Ensures that catalyst.adjoint accepts parameterized Python functions as arguments."""
-
-    def func(w, theta1, theta2, theta3=1):
-        qml.RX(theta1 * pnp.pi / 2, wires=w)
-        qml.RY(theta2 / 2, wires=w)
-        qml.RZ(theta3, wires=1)
-
-    device = qml.device(backend, wires=2)
-
-    @qjit
-    @qml.qnode(device)
-    def C_workflow(w, theta):
-        qml.PauliX(wires=0)
-        C_adjoint(func)(w, theta, theta2=theta)
-        qml.PauliY(wires=0)
-        return qml.state()
-
-    @qml.qnode(device)
-    def PL_workflow(w, theta):
-        qml.PauliX(wires=0)
-        PL_adjoint(func)(w, theta, theta2=theta)
-        qml.PauliY(wires=0)
-        return qml.state()
-
-    actual = C_workflow(w, p)
-    desired = PL_workflow(w, p)
-    assert_allclose(actual, desired)
-
-
-def test_adjoint_nested_fun(backend):
-    """Ensures that catalyst.adjoint allows arbitrary nesting."""
-
-    def func(A, I):
-        qml.RX(I, wires=1)
-        qml.RY(I, wires=1)
-        if I < 5:
-            I = I + 1
-            A(partial(func, A=A, I=I))()
-
-    @qjit
-    @qml.qnode(qml.device(backend, wires=2))
-    def C_workflow():
-        qml.RX(pnp.pi / 2, wires=0)
-        C_adjoint(partial(func, A=C_adjoint, I=0))()
-        qml.RZ(pnp.pi / 2, wires=0)
-        return qml.state()
-
-    @qml.qnode(qml.device("default.qubit", wires=2))
-    def PL_workflow():
-        qml.RX(pnp.pi / 2, wires=0)
-        PL_adjoint(partial(func, A=PL_adjoint, I=0))()
-        qml.RZ(pnp.pi / 2, wires=0)
-        return qml.state()
-
-    assert_allclose(C_workflow(), PL_workflow())
-
-
-def test_adjoint_qubitunitary(backend):
-    """Ensures that catalyst.adjoint supports QubitUnitary oprtations."""
-
-    def func():
-        qml.QubitUnitary(
-            jnp.array(
-                [
-                    [0.99500417 - 0.09983342j, 0.0 + 0.0j, 0.0 + 0.0j, 0.0 + 0.0j],
-                    [0.0 + 0.0j, 0.99500417 + 0.09983342j, 0.0 + 0.0j, 0.0 + 0.0j],
-                    [0.0 + 0.0j, 0.0 + 0.0j, 0.99500417 + 0.09983342j, 0.0 + 0.0j],
-                    [0.0 + 0.0j, 0.0 + 0.0j, 0.0 + 0.0j, 0.99500417 - 0.09983342j],
-                ]
-            ),
-            wires=[0, 1],
-        )
+from catalyst import adjoint, cond, for_loop, qjit, while_loop
 
-    verify_catalyst_adjoint_against_pennylane(func, qml.device(backend, wires=2))
 
+class CatalystTests:
```

Comment:
I cannot test this class locally without the following change,
```suggestion
class TestCatalyst:
```

Reply:
Good catch, thanks!